src/main/webapp/script.js

```diff
@@ -22,10 +22,13 @@ function passQuery() {
   document.getElementById('result').style.display = 'block';
   const information = document.getElementById('more-info');
   information.innerText = 'Please wait. Loading...';
+
   const query = new FormData(document.getElementById('query-form'));
   const personType = query.get('person-type');
   const action = query.get('action');
-  const location = query.get('location');
+  const locationName = query.get('location');
+  const location = document.querySelector("#location option[value='"+locationName+"']").dataset.value;
```

Comment:
I think with datalist you can type a value not in the list so you need some error checking here

Reply:
see https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_datalist, you can type random text in the box and submit and it will still send it through

Reply:
This is true, but I disabled that functionality in the HTML. The input of type list (that the datalist belongs to) has a pattern defined, so it only accepts answers that fit the pattern. The pattern is just all of the possible answers to the datalist separated by | (or symbols). It's a bit inelegant, so definitely open to alternate suggestions, but I do like that it rejects incorrect answers in the front end itself.